### ISSUE TITLE
Crash with extension 'Json to dart model' installed

### ISSUE THREAD

**Issue by Totodore** [evald24/vscode-extensions-profiles#13]
Hi !

I found a little bug that make this wonderful extension crash when the extension `json-to-dart` is installed when we want to create a profile. It comes from the following function in the utils.ts file:
```ts
export async function getAllExtensions() {
  const extPath = getExtensionsPath();
  let extensions: ExtensionValue[] = [];
  let obsolete: string[] = []; // default value

  if (fs.existsSync(extPath + ".obsolete")) obsolete = Object.keys(loadJSON(extPath + ".obsolete"));

  let all = await readdir(extPath);

  await Promise.all(
    all.map(async (name) => {
      if ((await stat(extPath + platform_slash + name)).isDirectory() && !obsolete.includes(name)) {
        let info: PackageJson = require(extPath + platform_slash + name + platform_slash + "package.json");
        if (!info.__metadata?.id)
          console.log(info);
        extensions.push({
          id: `${info.publisher.toLowerCase()}.${info.name.toLowerCase()}`,
          uuid: info.__metadata.id,
          label: info.displayName || info.name,
          description: info.description,
        });
      }
    }),
  );

  return extensions.sort((a: any, b: any) => {
    // eslint-disable-next-line curly
    if (a.label > b.label) return -1;
    // eslint-disable-next-line curly
    else if (a.label < b.label) return 1;
    // eslint-disable-next-line curly
    else return 0;
  });
}
```

![image](https://user-images.githubusercontent.com/26095587/131395779-cd81cac3-0dd1-4138-8c9b-cb78bdbe33f2.png)
It crashes at line 214 because the `__metadata` object is not present into the `info` object for the extension `json-to-dart`. 

I propose that if the use of this __metadata object is mandatory for the extension then that the extensions which don't have it are skipped. Meanwhile I'm going to uninstall `json-to-dart`.

**Comment by Totodore** [comment]
After uninstalling this extension it appears that the issue persist (with the same extension whereas I just removed it). So maybe there is another cause to this non existence of the `__metadata` object.

**Comment by evald24** [comment]
@Totodore Thank you, I will definitely consider improving the extension in the near future, I have no free time right now.

**Comment by evald24** [comment]
@Totodore I am very sorry that I could not address this issue earlier. The bug has fixed in version 1.0.9
